# Hypothesis Testing

Hypothesis testing is a statistical technique utilized to evaluate assertions or hypotheses about a population through the examination of sample data. The primary objective is to determine if there's enough evidence to support or reject a claim regarding a particular population parameter.

This process generally involves two hypotheses:

* the null hypothesis (H0), and
* the alternative hypothesis (H1).

In [1]:
#run first the library dont forget

import pandas as pd
from scipy import stats
from statsmodels.stats import weightstats as stests

In [2]:
#load the dataset

df = pd.read_csv('Golf.csv')

In [11]:
#preview the dataset

df[['Current','New']].describe()
df.head(5)

Current  New
0      264  277
1      261  269
2      267  263
3      272  266
4      258  262

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Data columns (total 2 columns):
 #   Column   Non-Null Count  Dtype
---  ------   --------------  -----
 0   Current  40 non-null     int64
 1   New      40 non-null     int64
dtypes: int64(2)
memory usage: 772.0 bytes


In [5]:
#look how many data (column and row)

df.shape

(40, 2)

## **Paired sample t-test:**

* The paires sample t-test is also called dependent sample-t test
* Its an uni-variate that test for a significant difference between 2 related variables.

Here's an example, to see if the new coating does have a significant effect on driving range.

H0: is there's no significant difference between the distance traveled by the current ball without coating and the new ball with coating

H1: is there's a significant difference between the distance traveled by the current ball without coating and the new ball with coating

In [12]:
ttest,pval = stats.ttest_rel(df['Current'], df['New'])
print(pval)

0.20916361823147053


P-value is a value that shows how strong the evidence is for the null hypothesis in a statistical test. This calculation produces a p-value of 0.20916361823147053.

Alpha is a significance level that indicates the limit for accepting or rejecting the null hypothesis in statistical analysis. Commonly used alpha values are 0.05 or 5%, although sometimes other values such as 0.01 or 0.10 may also be used depending on the context.

**But, we want to use an alpha of 0.05 as the tolerance error limit.**

In [14]:
if pval<0.05:
    print("reject null hypothesis")
else:
    print("accept null hypothesis")

accept null hypothesis


After we use an alpha of 5 percent or 0.05, it turns out that the p-value is 0.209 > alpha 0.05, so that means we accept H0, which means there's no significant difference between the distance traveled by the current ball without coating and the new ball with coating.